lib/autokey/scripting/engine.py

```diff
@@ -155,13 +202,20 @@ def create_phrase(self, folder, name: str, contents: str,
                 p.set_hotkey(*hotkey)
             if window_filter:
                 p.set_window_titles(window_filter)
+            # XXX: Could these next 3 be refactored to just set p.<val> to the
```

Comment:
The function is annotated to accept booleans. But it does not type check the input, so the user may pass anything truth-y or false-y, like `[1]` and `[]`. This if implicitly calls bool().

I’d say yes to your question, it can be refactored the way you proposed. It should be even better, because that is more explicit.
But then wrap the user passed value in bool(), like `p.prompt = bool(always_prompt)`. Otherwise a list or string (or whatever else user provided cruft) ends up in the data structure and later in the json serializer. I already discovered issues with this kind of issue when I tried to store function objects in the `store` API.

Reply:
Good catch. I'll do that.